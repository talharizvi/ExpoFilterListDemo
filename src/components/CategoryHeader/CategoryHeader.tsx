import React from "react";
import {Text, View, StyleSheet} from 'react-native';

interface Category {
    category: String,
    testID?: string;
}

const CategoryHeader: React.FC<Category>=({category})=>{

    return(
    <View testID='category-header'>
        <Text style={styles.categoryText}>{category}</Text>
    </View>
    );

}

const styles = StyleSheet.create({
    categoryText:{
        fontWeight:'bold',
        fontSize: 18,
        marginTop: 16,
        marginBottom:8
    }    
});

export default CategoryHeader