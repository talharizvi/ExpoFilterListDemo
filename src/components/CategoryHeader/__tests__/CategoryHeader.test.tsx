import React from 'react';
import { render, screen } from '@testing-library/react-native';
import CategoryHeader from '../CategoryHeader';

const category = 'Test Category';

describe('CategoryHeader component', () => {

  it('renders category text with category prop', () => {
    
    render(<CategoryHeader category={category} />);
    expect(screen.getByText(category)).toBeDefined();
  });

});