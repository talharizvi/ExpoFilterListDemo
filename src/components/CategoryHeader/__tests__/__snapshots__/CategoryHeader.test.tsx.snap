// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryHeader component renders category text with category prop 1`] = `
<View
  testID="category-header"
>
  <Text
    style={
      {
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 8,
        "marginTop": 16,
      }
    }
  >
    Test Category
  </Text>
</View>
`;
